tests/test_infection_dynamics.py

```diff
@@ -1229,3 +1228,28 @@ def test_infectiousness_multiplier_transmissions_increase_with_multiplier( self,
         is_trans_cnt_increasing = avg_trans.diff()[1:] > 0
 
         np.testing.assert_equal( np.all(is_trans_cnt_increasing), True, "Infectiousness does not increase with multiplier" )
+
+        
+    def test_waning_immunity_multiple_infections(self):
+        """
+        Test individuals have multiple infection events when transitions from recovered to
+        susceptible decrease exponentially (with a mean time of 25 days) after 15 days of being
+        fully immune.  
+        """
+
+        params = ParameterSet(constant.TEST_DATA_FILE, line_number = 1)
+        params = utils.turn_off_interventions(params, int(params.get_param("end_time")))
+        params.set_param("mean_time_to_susceptible_after_shift", 25)
+        params.set_param("time_to_susceptible_shift", 15)
+        params.set_param("end_time", 100)
+
+        params.write_params(constant.TEST_DATA_FILE)
+        file_output = open(constant.TEST_OUTPUT_FILE, "w")
+        completed_run = subprocess.run([constant.command], stdout = file_output, shell = True)
+
+        df_trans = pd.read_csv(constant.TEST_TRANSMISSION_FILE)
+        df_indiv = pd.read_csv(constant.TEST_INDIVIDUAL_FILE)
+        df_indiv = pd.merge(df_indiv, df_trans, 
+            left_on = "ID", right_on = "ID_recipient", how = "left")
+
+        np.testing.assert_equal(np.any(df_indiv.infection_count > 1), True)
```

Comment:
You should have a check on the distribution of time from recovered to susceptible.